Modules.md

```diff
@@ -140,6 +140,14 @@ The linear memory section may also contain an optional hint declaring the expect
 maximum heap usage. This hint is not semantically visible but can help a
 WebAssembly engine to optimize `grow_memory`.
 
+The linear memory section may optionally declare that the instance's
+linear memory is *externally aliasable*. How linear memory is aliased is up
+to the host environment (as with all module exports). The 
+[Web](Web.md#aliasing-linear-memory-from-JS) would reflect exported linear
+memory to JS as an `ArrayBuffer`. The MVP does not currently provide for
+*importing* linear memory though this may be added 
+[in the future](FutureFeatures.md#importing-linear-memory).
```

Comment:
I'm pretty wary of how this will interact with multi-threading and SAB. In C++ it's a bit of a fiasco to use shared memory:
* `volatile` should be used to denote external modification, so the compiler remains conservative.
* `std::atomic` should be used to enforce ordering and no-tear (yes, in addition to `volatile`).
* Non-lock-free atomics are guaranteed to work in an address-free way *within the same process* but definitely do *not* work when using shared memory across processes.

This makes me wary of specifying things as you've outlined here without properly understanding what this implies for threads in WebAssembly. Does it entirely prevent optimizations? Valid reordering? 64-bit atomics on some platforms?

I'm OK specifying this as you've done fi you think it'll move stuff forward, but I insist on opening an issue with the above text, to revisit *before* MVP so we're not stuck when we want to add threads.

WDYT?